base_local_planner/src/goal_functions.cpp

```diff
@@ -113,9 +113,10 @@ namespace base_local_planner {
       geometry_msgs::TransformStamped plan_to_global_transform = tf.lookupTransform(global_frame, ros::Time(),
           plan_pose.header.frame_id, plan_pose.header.stamp, plan_pose.header.frame_id, ros::Duration(0.5));
 
-      //let's get the pose of the robot in the frame of the plan
-      geometry_msgs::PoseStamped robot_pose;
-      tf.transform(global_pose, robot_pose, plan_pose.header.frame_id);
+      //let's get the pose of the robot in the frame of the plan using the latest transform available
+      geometry_msgs::PoseStamped robot_pose = global_pose;
+      robot_pose.header.stamp = ros::Time();
+      tf.transform(robot_pose, robot_pose, plan_pose.header.frame_id);
```

Comment:
looks a bit hacky, might be cleaner to just use an overload that exposes the time (needs a fixed frame though): http://docs.ros.org/en/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1BufferInterface.html#a99961ac55e19b08d2961d999a458d9ad

Reply:
I think it's the name of a fixed frame (i.e. a frame that does not move / change over time)? Examples of this would be map, world, odom, etc.